Build project is failing to build due to targeting netcoreapp3.1 which is no longer supported by windows-latest image

## Describe the bug
The Build project that is included in the template files is targeting the old netcoreapp3.1 framework. The framework is no longer installed by default on the 'windows-latest' image, and therefore the pipeline fails.

## To reproduce
Attempt to execute the build.ps1 file in a pipeline using the 'windows-latest' image.

## Expected behaviour
For the build to succeed.

## Screenshots
`Starting: Build package
==============================================================================
Task         : PowerShell
Description  : Run a PowerShell script on Linux, macOS, or Windows
Version      : 2.220.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
==============================================================================
Generating script.
Formatted command: . 'D:\a\1\s\build.ps1' --target compile --solution-type managed
========================== Starting Command Output ===========================
"C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'D:\a\_temp\18a9a8b7-89d4-446e-8ea1-6ba1e2c935a9.ps1'"
PowerShell Desktop version 5.1.20348.1366
Microsoft (R) .NET Core SDK version 7.0.304
You must install or update .NET to run this application.

App: D:\a\1\s\build\bin\Debug\_build.exe
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '3.1.0' (x64)
.NET location: C:\Program Files\dotnet

The following frameworks were found:
  6.0.5 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  6.0.18 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  6.0.19 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  7.0.7 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

Learn about framework resolution:
https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=3.1.0&arch=x64&rid=win10-x64
##[error]PowerShell exited with code '1'.
Finishing: Build package
`

--- a/__tests__/sample-with-solution/.vscode/launch.json
+++ b/__tests__/sample-with-solution/.vscode/launch.json
@@ -10,7 +10,7 @@
             "request": "launch",
             "preLaunchTask": "build",
             // If you have changed target frameworks, make sure to update the program path.
-            "program": "${workspaceFolder}/build/bin/Debug/netcoreapp3.1/_build.dll",
+            "program": "${workspaceFolder}/build/bin/Debug/_build.dll",
             "args": [],
             "cwd": "${workspaceFolder}/build",
             // For more information about the 'console' field, see https://aka.ms/VSCode-CS-LaunchJson-Console

--- a/__tests__/sample-with-solution/build/_build.csproj
+++ b/__tests__/sample-with-solution/build/_build.csproj
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net7.0</TargetFramework>
     <RootNamespace></RootNamespace>
     <NoWarn>CS0649;CS0169</NoWarn>
     <NukeRootDirectory>..</NukeRootDirectory>

--- a/src/generators/app/templates/source/.vscode/launch.json
+++ b/src/generators/app/templates/source/.vscode/launch.json
@@ -10,7 +10,7 @@
             "request": "launch",
             "preLaunchTask": "build",
             // If you have changed target frameworks, make sure to update the program path.
-            "program": "${workspaceFolder}/build/bin/Debug/netcoreapp3.1/_build.dll",
+            "program": "${workspaceFolder}/build/bin/Debug/_build.dll",
             "args": [],
             "cwd": "${workspaceFolder}/build",
             // For more information about the 'console' field, see https://aka.ms/VSCode-CS-LaunchJson-Console

--- a/src/generators/app/templates/source/build/_build.csproj
+++ b/src/generators/app/templates/source/build/_build.csproj
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net7.0</TargetFramework>
     <RootNamespace></RootNamespace>
     <NoWarn>CS0649;CS0169</NoWarn>
     <NukeRootDirectory>..</NukeRootDirectory>

